### ISSUE TITLE
（GAS）在庫管理システム国内用 

### ISSUE THREAD

**Issue by ryuji0909** [ryuji0909/memo#2]
`function doGet() {
  var template = HtmlService.createTemplateFromFile('index');
  return template.evaluate();
}

function createTable(isFirstTable) {
  const id = "17KMA-kZ7PRLiHYR7_iOPwM2FVpxJ93w-YM0YZAIJuuk"; // スプレッドシートのIDを更新
  const ss = SpreadsheetApp.openById(id);
  const sheet = ss.getSheetByName("国内仕入先一覧"); // "国内仕入先一覧"シートを指定

  // 動的な範囲選択に変更。最終行と最終列を取得して範囲を指定します。
  const lastRow = sheet.getLastRow();
  const lastColumn = sheet.getLastColumn();
  const range = sheet.getRange(2, 1, lastRow - 1, lastColumn); // "A2"から最終行までの範囲を動的に指定
  const values = range.getValues();

  var result = "";
  var kana   = "";

  // ADD 2024/06/11 isFirstTableがfalseの場合にヘッダーを表示
  if (!isFirstTable) {
    result += '<table>';
    result += '<thead>';
    result += '<tr>';
    result += '<th>CODE</th> <th>仕入先名</th> <th>TEL</th> <th>FAX</th> <th>WEB-EDI</th> <th>標準リードタイム</th> <th>詳細ページ</th>';
    result += '</tr>';
    result += '</thead>';
    result += '</table>'; 
  }
  
  // スプレッドシートから取得したデータをループ
  for (var i = 0; i < values.length; i++) {
    if (kana !== values[i][0]) {
      kana = values[i][0];
      result += '<table class="supplier-table" data-kana="' + kana + '">';
    }
    
    let row = values[i];
    
    // rowの値が空白だった場合"n/a"と表記
    for (var j = 0; j < row.length; j++){
      if(row[j] == ""){
        row[j] = "n/a";
      }
    }

    const date = new Date(row[5]);
    const year = date.getFullYear();
    const month = ("0" + (date.getMonth() + 1)).slice(-2);
    const day = ("0" + date.getDate()).slice(-2);
    const formattedDate = year + "/" + month + "/" + day;

    // 数字を4桁で表示、2桁や3桁の場合には先頭に0を追加
    const code = row[1].toString().padStart(4, '0');

    result += '<td>' + code + '</td> <td>' + row[2] +'</td> <td>' + row[3] + '</td> <td>' + row[4] + '</td> <td>' + formattedDate + '</td> <td>' + row[6] + '</td>' + '<td><a href="https://script.google.com/a/macros/hq.stylinglife.co.jp/s/AKfycby83dJtABRL1KbrIv5brjPxLe1w3kDzzLk4_4QD833Lo-9h0mMVA2gmKA54lW7gHBHi/exec?p2=' + encodeURIComponent(row[1]) + '"target="_blank">詳細ページ</a></td>';

    result += '</tr>';
  }
 
  result += '</table>';
  // console.log(result);
  return result;
}`

**Comment by ryuji0909** [comment]
`<!DOCTYPE html>
<html>
<head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
<link rel="stylesheet" href="//code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
<script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
<style>
  /*ADD 2024年06/10* 新規登録リンク*/
  .link a {
    display: inline-block;
    text-decoration: none;
    color: black;
    font-size: 16px;
  }

  button {
    margin-left: 20px;
  }

  /*ADD 2024年06/10*　新規登録リンク*/
  .header {
    display: flex;
    align-items: center;
  }

  .list-container {
    width: 500px;
  }

  .list-container li {
    width: 50px;
    float: left;
    height: 20px;
  }

  .list-container li:nth-child(5n+1) {
    clear: left;
  }
        
  table{
    width: 75%;
    border-collapse:separate;
    border-spacing: 0;
  }

  table th:first-child{
    border-radius: 5px 0 0 0;
  }

  table th:last-child{
    border-radius: 0 5px 0 0;
    border-right: 1px solid #3c6690;
  }

  table th{
    text-align: center;
    color:white;
    background: linear-gradient(#829ebc,#225588);
    border-left: 1px solid #3c6690;
    border-top: 1px solid #3c6690;
    border-bottom: 1px solid #3c6690;
    box-shadow: 0px 1px 1px rgba(255,255,255,0.3) inset;
    width: 10%;
    padding: 10px 0;
  }

  table td{
    text-align: center;
    border-left: 1px solid #a8b7c5;
    border-bottom: 1px solid #a8b7c5;
    border-top:none;
    box-shadow: 0px -3px 5px 1px #eee inset;
    width: 10%;
    padding: 10px 0;
  }

  table td:last-child{
    border-right: 1px solid #a8b7c5;
  }

  table tr:last-child td:first-child {
    border-radius: 0 0 0 5px;
  }

  table tr:last-child td:last-child {
    border-radius: 0 0 5px 0;
  }

  h4 {
    color: red;
    font-size: 15px;
  }   
</style>
</head>
<body>
  <div class="header">
    <h1>仕入先コード表</h1> 
    <div class="link">
      <button><a href="https://script.google.com/a/macros/hq.stylinglife.co.jp/s/AKfycby83dJtABRL1KbrIv5brjPxLe1w3kDzzLk4_4QD833Lo-9h0mMVA2gmKA54lW7gHBHi/exec?p2=" target="_blank">新規登録</a></button>
    </div>
  </div>

  <h4>検索するときはCtrl ＋ Fで検索可能です。</h4>

  <div><ul id="kana-list" class="list-container">
    <li><a href="#" data-kana="あ">あ</a></li>
    <li><a href="#" data-kana="い">い</a></li>
    <li><a href="#" data-kana="う">う</a></li>
    <li><a href="#" data-kana="え">え</a></li>
    <li><a href="#" data-kana="お">お</a></li>

    <li><a href="#" data-kana="か">か</a></li>
    <li><a href="#" data-kana="き">き</a></li>
    <li><a href="#" data-kana="く">く</a></li>
    <li><a href="#" data-kana="け">け</a></li>
    <li><a href="#" data-kana="こ">こ</a></li>

    <li><a href="#" data-kana="さ">さ</a></li>
    <li><a href="#" data-kana="し">し</a></li>
    <li><a href="#" data-kana="す">す</a></li>
    <li><a href="#" data-kana="せ">せ</a></li>
    <li><a href="#" data-kana="そ">そ</a></li>

    <li><a href="#" data-kana="た">た</a></li>
    <li><a href="#" data-kana="ち">ち</a></li>
    <li><a href="#" data-kana="つ">つ</a></li>
    <li><a href="#" data-kana="て">て</a></li>
    <li><a href="#" data-kana="と">と</a></li>

    <li><a href="#" data-kana="な">な</a></li>
    <li><a href="#" data-kana="に">に</a></li>
    <li><a href="#" data-kana="ぬ">ぬ</a></li>
    <li><a href="#" data-kana="ね">ね</a></li>
    <li><a href="#" data-kana="の">の</a></li>

    <li><a href="#" data-kana="は">は</a></li>
    <li><a href="#" data-kana="ひ">ひ</a></li>
    <li><a href="#" data-kana="ふ">ふ</a></li>
    <li><a href="#" data-kana="へ">へ</a></li>
    <li><a href="#" data-kana="ほ">ほ</a></li>

    <li><a href="#" data-kana="ま">ま</a></li>
    <li><a href="#" data-kana="み">み</a></li>
    <li><a href="#" data-kana="む">む</a></li>
    <li><a href="#" data-kana="め">め</a></li>
    <li><a href="#" data-kana="も">も</a></li>

    <li><a href="#" data-kana="や">や</a></li>
    <li><a href="#" data-kana="　">　</a></li>
    <li><a href="#" data-kana="ゆ">ゆ</a></li>
    <li><a href="#" data-kana="　">　</a></li>
    <li><a href="#" data-kana="よ">よ</a></li>

    <li><a href="#" data-kana="ら">ら</a></li>
    <li><a href="#" data-kana="り">り</a></li>
    <li><a href="#" data-kana="る">る</a></li>
    <li><a href="#" data-kana="れ">れ</a></li>
    <li><a href="#" data-kana="ろ">ろ</a></li>

    <li><a href="#" data-kana="わ">わ</a></li>
    <li><a href="#" data-kana="　">　</a></li>
    <li><a href="#" data-kana="を">を</a></li>
    <li><a href="#" data-kana="　">　</a></li>
    <li><a href="#" data-kana="ん">ん</a></li>
     </ul></div> 
    <br><br><br>
    <br><br><br><br><br><br><br>

    <div id="createdTable"></div>

<script>

  // サーバからデータを取得し、テーブルを作成する
  function createInnerTable() {
    google.script.run
      .withSuccessHandler(function(result) {
        $('#createdTable').html(result);
          supplierTables = document.querySelectorAll('.supplier-table');
      })
  .createTable();
  }
  window.addEventListener('load', createInnerTable);

  let supplierTables;

  // 「かな」がクリックされた場合の処理
  document.querySelector('#kana-list').addEventListener('click', (event) => {
    // クリックされた要素がaタグであるか判定
    if (event.target.tagName === 'A') {
      event.preventDefault();
        // クリックされた値を取得
        const kana = event.target.getAttribute('data-kana');
        console.log(supplierTables)
        // supplierTablesに要素があるかどうか判定
        if (supplierTables && supplierTables.length > 0) { 
          // 1つ1つ取り出し、テーブルに対してCSSを付加させる 
          supplierTables.forEach((table) => {
            table.style.display = 'none';
          });

          // クリックされた「かな」文字に対応する一覧を表示
          const supplierTable = document.querySelector(`.supplier-table[data-kana="${kana}"]`);
            if (supplierTable) {
                supplierTable.style.display = 'block';
             }
          }
     
            /*ADD 2024/06/11 クリックされたらフラグを持たせてサーバー側に渡す。*/
            google.script.run.createTable(false);
    };      
  });
</script>
</body>
</html>`
